src/ol/interaction/selectinteraction.js

```diff
@@ -211,6 +234,27 @@ ol.interaction.Select.prototype.getFeatures = function() {
 
 
 /**
+ * Returns the associated {@link ol.layer.Vector vectorlayer} of
+ * the (last) selected feature.
+ * @param {ol.Feature} feature Feature
+ * @return {ol.layer.Vector} Layer.
+ * @api
+ */
+ol.interaction.Select.prototype.getLayer = function(feature) {
+  goog.asserts.assertInstanceof(feature, ol.Feature,
+      'feature should be an ol.Feature');
+  var key = goog.getUid(feature).toString();
```

Comment:
If you implement my suggestion from above, then all it would take to get the layer for the feature would be
```js
return this.featureLayerAssociation_[key];
```